src/Service/Search.php

```diff
@@ -33,4 +33,23 @@ public function stationsSearch(string $searchTerm)
 
         return $foundStations;
     }
+
+    public function stationsSearchRawResults(string $searchTerm)
+    {
+        // flatten search results
+        $foundStations = array_merge(...array_values($this->stationsSearch($searchTerm)));
+        //de-duplicate stations array
+        $stations = [];
+        $ids = [];
+        foreach ($foundStations as $station) {
+            $id = $station->getId();
+
+            if (!in_array($id, $ids)) {
```

Comment:
You should use isset() instead of in_array(). isset is faster than in_array (like a lot more)